addons/website_links/static/src/interactions/website_links_charts.js

```diff
@@ -0,0 +1,262 @@
+import { Interaction } from "@web/public/interaction";
+import { _t } from "@web/core/l10n/translation";
+import { deserializeDate } from "@web/core/l10n/dates";
+import { registry } from "@web/core/registry";
+import { loadBundle } from "@web/core/assets";
+const { DateTime } = luxon;
+
+class WebsiteLinksCharts extends Interaction {
+    static selector = ".o_website_links_chart";
+    dynamicContent = {
+        "#all_time_clicks_chart .title": {
+            "t-out": () => `${this.numAllTimeClicks} ${_t("clicks")}`,
+        },
+        "#last_month_clicks_chart .title": {
+            "t-out": () => `${this.numLastMonthClicks} ${_t("clicks")}`,
+        },
+        "#last_week_clicks_chart .title": {
+            "t-out": () => `${this.numLastWeekClicks} ${_t("clicks")}`,
+        },
+    };
+
+    setup() {
+        this.orm = this.services.orm;
+        const linkId = parseInt(this.el.querySelector("#link_id").value);
+        this.linksDomain = ["link_id", "=", linkId];
+        this.totalClicks = [];
+        this.clicksByDay = [];
+        this.pieChartsData = [
+            {
+                selector: "#all_time_countries_charts",
+                fetch: this.getClicksByCountry.bind(this),
+                data: [],
+                numCountries: 0,
+            },
+            {
+                selector: "#last_month_countries_charts",
+                fetch: this.getLastMonthClicksByCountry.bind(this),
+                data: [],
+                numCountries: 0,
+            },
+            {
+                selector: "#last_week_countries_charts",
+                fetch: this.getLastWeekClicksByCountry.bind(this),
+                data: [],
+                numCountries: 0,
+            },
+        ];
+        const width = this.el.querySelector(this.pieChartsData[0].selector).scrollWidth;
+        for (const pieChartData of this.pieChartsData) {
+            // Adapt height of chart for when many country labels
+            this.dynamicContent[pieChartData.selector] = {
+                "t-att-style": () => ({
+                    height: `${Math.max(pieChartData.data.length * (width > 750 ? 1 : 2), 20)}em`,
+                }),
+            };
+            this.dynamicContent[`${pieChartData.selector} .title`] = {
+                "t-out": () => `${pieChartData.numCountries} ${_t("countries")}`,
+            };
+        }
```

Comment:
Oh woaw... don't do that. You actually get the selected `el` as the parameter of `t-att-style` or `t-out`, so you can write everything in the predefined `dynamicContent`.
If needed maybe create a map in `this` to find the right data for a given chart, e.g. based on the id field.